lib/core/src/Cardano/Wallet.hs

```diff
@@ -901,14 +907,19 @@ readNextWithdrawal
     :: forall ctx s t k.
         ( HasDBLayer s k ctx
         , HasTransactionLayer t k ctx
+        , HasNetworkLayer t ctx
+        , HasRewardAccount s k
+        , HasRewardAccount s ShelleyKey
         )
     => ctx
     -> WalletId
-    -> IO (Quantity "lovelace" Word64)
-readNextWithdrawal ctx wid = db & \DBLayer{..} -> do
-    (pp, withdrawal) <- atomically $ (,)
-        <$> readProtocolParameters pk
-        <*> fmap getQuantity (readDelegationRewardBalance pk)
+    -> Either WalletId SomeMnemonic
+    -> ExceptT ErrFetchRewards IO (Quantity "lovelace" Word64)
+readNextWithdrawal ctx wid src = db & \DBLayer{..} -> do
```

Comment:
`src` was unexpected name for me. tbh. maybe `acctSrc`, `chimericOrigin` ?

Reply:
Good point. My creativity for names at 4 a.m. was somewhat diminished :joy: